src/other/utils.jl

```diff
@@ -417,3 +417,21 @@ function _findall(B::BitVector)::Union{UnitRange{Int}, Vector{Int}}
     end
     @assert false "should not be reached"
 end
+
+function _copy_metadata!(dst::DataFrame, src; mode::Symbol)
+    # TODO: define other allowed modes when needed
+    # currently :overwrite is only added as it was only needed at this stage
+    @assert mode == :overwrite
```

Comment:
I wonder whether separate functions would be more appropriate. Hard to tell without seeing what we actually need and how it could be implemented though.